distributed/shuffle/_worker_extension.py

```diff
@@ -78,10 +83,16 @@ def __init__(
         self.scheduler = scheduler
         self.closed = False
 
-        self._disk_buffer = DiskShardsBuffer(
-            directory=directory,
-            memory_limiter=memory_limiter_disk,
-        )
+        self._disk_buffer: FileShardsBuffer
+        # TODO: This needs rethinking because it is run on the worker
+        # where the config may be out of sync.
```

Comment:
I'm fine with this limitation. There are many config options that need to be set on the worker, e.g. everything around memory management